mpcforces_extractor/playground/database.py

```diff
@@ -0,0 +1,191 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict
+from fastapi import HTTPException
+from sqlmodel import Session, create_engine, SQLModel, Field, select, Column, JSON
+from mpcforces_extractor.datastructure.rigids import MPC
+from mpcforces_extractor.datastructure.rigids import MPC_CONFIG
+from mpcforces_extractor.datastructure.entities import Node, Element
+
+
+class MPCDBModel(SQLModel, table=True):
+    """
+    Database Representation of MPC Class
+    """
+
+    id: int = Field(primary_key=True)
+    config: str = Field()  # Store MPC_CONFIG as a string
+    master_node: int = Field()  # Store master node as an integer
+    nodes: str = Field()  # Store nodes as a string
+    part_id2nodes: Dict = Field(
+        default_factory=dict, sa_column=Column(JSON)
+    )  # Store part_id2nodes as a dictionary
+
+    def to_mpc(self):
+        """
+        Method to convert MPCDBModel back to MPC object if needed
+        """
+        print(f"Converting MPCDBModel to MPC: id={self.id}, nodes={self.nodes}")
+        nodes_list = (
+            str(self.nodes).split(",") if self.nodes else []
+        )  # Add a check to avoid splitting None
+        mpc = MPC(
+            element_id=self.id,
+            mpc_config=MPC_CONFIG[self.config],  # Convert string back to enum
+            master_node=Node.node_id2node[
+                self.master_node
+            ],  # Handle node conversion as needed
+            nodes=[Node.node_id2node[int(node_id)] for node_id in nodes_list],
+            dofs="",
+        )
+        mpc.part_id2node_ids = self.part_id2nodes
+        return mpc
+
+
+class NodeDBModel(SQLModel, table=True):
+    """
+    Database Representation of Node Instance
+    """
+
+    id: int = Field(primary_key=True)
+    coord_x: float = Field()
+    coord_y: float = Field()
+    coord_z: float = Field()
+
+    def to_node(self):
+        """
+        Method to convert NodeDBModel back to Node object if needed
+        """
+        return Node(node_id=self.id, coords=[self.coord_x, self.coord_y, self.coord_z])
+
+
+class MPCForcesExtractorDatabase(ABC):
+    """
+    This class represents a generic MPC database.
+    """
+
+    @abstractmethod
+    async def get_mpcs(self) -> List[MPCDBModel]:
+        """
+        Get all MPCs
+        """
+        raise NotImplementedError
+
+
+class FakeDatabase(MPCForcesExtractorDatabase):
```

Comment:
For a demonstration not bad but no use in the general code.
Thats missing it by an inch imho, as the database would work quite well without the init() being fake. The rest seems legit to me also for the other code.